swagger_spec_validator/validator20.py

```diff
@@ -156,6 +156,63 @@ def validate_apis(apis, deref):
             validate_unresolvable_path_params(api_name, all_path_params)
 
 
+def get_collapsed_properties_type_mapping(definition, deref):
+    definition = deref(definition)
+    required_properties = {}
+    not_required_properties = {}
+    if definition.get('allOf'):
+        for inner_definition in definition['allOf']:
+            inner_required_properties, inner_not_required_properties = get_collapsed_properties_type_mapping(inner_definition, deref)
+            required_properties.update(inner_required_properties)
+            not_required_properties.update(inner_not_required_properties)
+    else:
+        properties = {
+            prop_name: prop_schema.get('type', 'object')
+            for prop_name, prop_schema in iteritems(definition.get('properties', {}))
+        }
+        required_properties_set = set(definition.get('required', []))
+
+        required_properties = {
+            k: v
+            for k, v in iteritems(properties)
+            if k in required_properties_set
+        }
+        not_required_properties = {
+            k: v
+            for k, v in iteritems(properties)
+            if k not in required_properties_set
+        }
+    return required_properties, not_required_properties
```

Comment:
Returning tuples is not the most descriptive way of doing this, but I guess using a `dict` or `namedtuple` adds quite a bit of code especially to the recursion. At least the function should be named get_collapsed_properties_type_mapping**s**, as it returns two mappings.